src/components/UpdatePassword/UpdatePasswordButton.tsx

```diff
@@ -0,0 +1,129 @@
+import { useContext, useState } from "react";
+import { updatepasswordService } from "../../services/auth/updatepassword.service";
+import { FieldValues, useForm } from "react-hook-form";
+import { AuthContext } from "../../context/AuthContext";
+import toast from "react-hot-toast";
+
+export function UpdatePassword() {
+  const { session } = useContext(AuthContext);
+  const [showModal, setShowModal] = useState(false);
+  const {
+    register,
+    handleSubmit,
+    formState: { errors }
+  } = useForm();
+
+  interface UpdatepasswordRequest {
+    oldPassword: string;
+    newPassword: string;
+    token: string;
+  }
+
+  const onSubmit = async (formData: FieldValues, token: string) => {
+    const req: UpdatepasswordRequest = {
+      oldPassword: formData.password,
+      newPassword: formData.confirmPassword,
+      token: token
+    };
+    if (!errors.password && !errors.confirmPassword) {
+      try {
+        const response = await updatepasswordService(req);
+        if (response.success === true) {
+          toast.success("Password updated successfully");
+        } else {
+          toast.error("Failed to update password");
+        }
+      } catch (error) {
+        toast.error("Failed to update password");
+      }
+    }
+  };
+
+  return (
+    <>
+      <button
+        className="w-full rounded-lg border border-blue-600 bg-white p-2 text-center text-lg font-bold text-blue-600"
+        onClick={() => setShowModal(true)}
+      >
+        Update Password
+      </button>
+      {showModal ? (
+        <div className="absolute top-full mt-4 flex w-4/5 flex-col items-center justify-center gap-4 border bg-white p-4">
```

Comment:
The modal is huge. It's supposed to be a square-shaped modal according to your own designs.

![image](https://github.com/hawks-atlanta/frontend-react/assets/62714297/4624e570-13e2-401b-8311-2fd9c2bab7fe)

Reply:
Actually, there is a [dialog component](https://github.com/hawks-atlanta/frontend-react/blob/dev/src/components/Dialog.tsx) in this codebase; use it.

Reply:
![image](https://github.com/hawks-atlanta/frontend-react/assets/131629166/e86f859e-ba3f-4e35-9867-91c02a0262d1)